.github/workflows/release.yml

```diff
@@ -39,4 +39,13 @@ jobs:
           SONATYPE_KEY: ${{ secrets.SONATYPE_KEY }}
           GPG_PRIVATE_KEY: ${{ secrets.GPG_PRIVATE_KEY }}
           GPG_PASSWORD: ${{ secrets.GPG_PASSWORD }}
+      - name: Create release tag for release builds
+        env:
+          GH_TOKEN: ${{ secrets.OPENTELEMETRYBOT_GITHUB_TOKEN }}
+        if: "${{ inputs.release_type == 'final' }}"
+        run: |
+          VERSION_NUMBER=$(grep ^version= gradle.properties  | sed -e 's/^.*=//')
```

Comment:
Not for this PR but just adding a suggestion, upgrading the `gradle.properties` and creating the GH release could be part of the release automation as well.

`gradle.properties` could be upgraded with a bash script called in this workflow, and the version would be gotten from the `workflow_dispatch` as well.

The GH release could be created with an Action, eg [action-gh-release](https://github.com/softprops/action-gh-release) or similar.

Reply:
Yeah definitely! I think there's a ton of room for incremental improvements to the release process. The upstream repos have gotten quite sophisticated -- including generating the contributor list etc. 👍🏻

Reply:
See #157 as well. We're open for contributions. :)